eng/pipelines/agent-pool-migration.yml

```diff
@@ -0,0 +1,134 @@
+# The pipeline is mainly used to migrate our agent pool from one version to another version. 
```

Comment:
Just to be clear we should set triggers to none in this pipeline to indicate it is intended to be ran manually. Like https://github.com/Azure/azure-sdk-tools/blob/main/eng/pipelines/mirror-repos.yml#L3-L4